source/retryable-writes/retryable-writes.rst

```diff
@@ -405,7 +405,8 @@ If a retry attempt also fails, drivers MUST update their topology according to
 the SDAM spec (see: `Error Handling`_). If an error would not allow the caller
 to infer that an attempt was made (e.g. connection pool exception originating
 from the driver) or the error is labeled "NoWritesPerformed", the error from
-the previous attempt should be raised.
+the previous attempt should be raised. If all server errors are labeled
+"NoWritesPerformed", then the first error should be raised.
```

Comment:
> If all server errors are labeled "NoWritesPerformed", then the first error should be raised.

Is `previousError` in the code below intended to be the _first_ error encountered? If not, and `previousError` could be updated multiple times, the behavior below seems to conflict with this statement as the _first_ error is never permanently saved.

Reply:
Assuming that an error occurs, the idea is to always set "previousError" with the first error encountered, without condition. Otherwise, if the error is NOT labeled "NoWritesPerformed", then update the "previousError" to the "originalError":

```
if (originalError is not DriverException && ! originalError.hasErrorLabel("NoWritesPerformed") ||
        previousError == null) {
    previousError = originalError;
}
```

This behavior will "save" the first error only on the condition that the sequence of errors encountered are all labeled "NoWritesPerformed". Otherwise, a sequence of errors all labeled "NoWritesPerformed" will result in a "null" error.